Framework/Algorithms/src/CreateFloodWorkspace.cpp

```diff
@@ -0,0 +1,207 @@
+
+namespace {
+namespace Prop {
+std::string const FILENAME("Filename");
+std::string const OUTPUT_WORKSPACE("OutputWorkspace");
+std::string const START_X("StartSpectrumIndex");
+std::string const END_X("EndSpectrumIndex");
+std::string const EXCLUDE("Exclude");
+std::string const BACKGROUND("Background");
+} // namespace Prop
+
+double const VERY_BIG_VALUE = 10000.0;
+std::map<std::string, std::string> const funMap{
+    {"Linear", "name=LinearBackground"}, {"Quadratic", "name=Quadratic"}};
+
+} // namespace
+
+namespace Mantid {
+namespace Algorithms {
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(CreateFloodWorkspace)
+
+const std::string CreateFloodWorkspace::name() const {
+  return "CreateFloodWorkspace";
+}
+
+const std::string CreateFloodWorkspace::summary() const {
+  return "Algorithm to create a flood correction workspace for reflectometry "
+         "data reduction.";
+}
+
+int CreateFloodWorkspace::version() const { return 1; }
+
+const std::vector<std::string> CreateFloodWorkspace::seeAlso() const {
+  return {"ReflectometryReductionOneAuto"};
+}
+
+const std::string CreateFloodWorkspace::category() const {
+  return "Reflectometry\\ISIS";
+}
+
+/** Initialize the algorithm's properties.
+ */
+void CreateFloodWorkspace::init() {
+  const FacilityInfo &defaultFacility = ConfigService::Instance().getFacility();
+  std::vector<std::string> exts = defaultFacility.extensions();
+
+  declareProperty(make_unique<MultipleFileProperty>(Prop::FILENAME, exts),
+                  "The name of the flood run file(s) to read. Multiple runs "
+                  "can be loaded and added together, e.g. INST10+11+12+13.ext");
+
+  declareProperty(Prop::START_X, EMPTY_DBL(),
+                  "Start value of the fitting interval");
+  declareProperty(Prop::END_X, EMPTY_DBL(),
+                  "End value of the fitting interval");
+
+  declareProperty(Kernel::make_unique<ArrayProperty<double>>(Prop::EXCLUDE),
+                  "Spectra to exclude");
+
+  std::vector<std::string> allowedValues;
+  for (auto i : funMap)
+    allowedValues.push_back(i.first);
+  auto backgroundValidator =
+      boost::make_shared<ListValidator<std::string>>(allowedValues);
+  declareProperty(Prop::BACKGROUND, "Linear", backgroundValidator,
+                  "Background function.");
+
+  declareProperty(make_unique<WorkspaceProperty<MatrixWorkspace>>(
+                      Prop::OUTPUT_WORKSPACE, "", Direction::Output),
+                  "The flood correction workspace.");
+}
+
+MatrixWorkspace_sptr CreateFloodWorkspace::getInputWorkspace() {
+  std::string fileName = getProperty(Prop::FILENAME);
+  auto alg = createChildAlgorithm("Load", 0, 0.8);
+  alg->setProperty("Filename", fileName);
+  alg->setProperty("LoadMonitors", false);
+  alg->setProperty("OutputWorkspace", "dummy");
+  alg->execute();
+  Workspace_sptr ws = alg->getProperty("OutputWorkspace");
+  return boost::dynamic_pointer_cast<MatrixWorkspace>(ws);
```

Comment:
Have you considered that Load can return a group workspace, for example when the runs are listed R1, R2 ?